### PR TITLE
Feat add filtering flinksql

### PR DESCRIPTION
**Pull request by bbejeck**
### Description

[asana](https://app.asana.com/0/1201951209538300/1203941696282748/f)

### Staging Docs

http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/feat_add_filtering_flinksql/ 
http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/feat_add_filtering_flinksql/filter-a-stream-of-events/flinksql.html

### CODE REVIEW

**Comment by bbejeck** [comment]
@davetroiano - I'm aware the test probably won't pass, but the text is ready for review.  I'll address the test for this PR on Monday

**Comment by bbejeck** [comment]
Build passing for [this PR](https://semaphore.ci.confluent.io/jobs/520fb34d-3a66-4636-9c0e-6d5f2df13be7); there's an issue with the unit test, which I'll address, but otherwise, this is ready for review.

File: _includes/tutorials/filtering/flinksql/code/src/test/java/io/confluent/developer/FlinkSqlFilteringTest.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    // cause this INSERT to terminate once the latest offset of movie_sales table is reached.
+    streamTableEnv.executeSql(getResourceFileContents("populate-publication-events.sql"));
+
+    // execute query on result table that should have movie sales aggregated by release year
+    TableResult tableResult = streamTableEnv.executeSql(getResourceFileContents("query-publications-by-author.sql"));
+
+    // Compare actual and expected results. Convert result output to line sets to compare so that order
+    // doesn't matter, because the grouped result order doesn't matter -- 2017's could come before or after 2019's.
+    String actualTableauResults = tableauResults(tableResult);
+    String expectedTableauResults = getResourceFileContents("expected-expected-publications-by-author.txt");
```
Comment (davetroiano) [inline _includes/tutorials/filtering/flinksql/code/src/test/java/io/confluent/developer/FlinkSqlFilteringTest.java:33]: removing leading dupe `expected-`

File: _includes/tutorials/filtering/flinksql/code/src/test/java/io/confluent/developer/FlinkSqlFilteringTest.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+       Optional.of(kafkaPort),Optional.of(schemaRegistryPort))).await();
+    streamTableEnv.executeSql(getResourceFileContents("create-publications-by-author.sql.template",
+       Optional.of(kafkaPort),Optional.of(schemaRegistryPort))).await();
+  
+
+    // We can't call await() on this result since it won't return. In Flink 17 and later this can change
+    // by setting 'scan.bounded.mode' = 'latest-offset' in the movie_sales CREATE TABLE statement, which will
+    // cause this INSERT to terminate once the latest offset of movie_sales table is reached.
+    streamTableEnv.executeSql(getResourceFileContents("populate-publication-events.sql"));
+
```
Comment (davetroiano) [inline _includes/tutorials/filtering/flinksql/code/src/test/java/io/confluent/developer/FlinkSqlFilteringTest.java:26]: also need to execute `populate-publication-events-by-author.sql`

**Review by davetroiano — APPROVED** [review]
nice job - a few minor typos and unit test fixes needed

File: _includes/tutorials/filtering/flinksql/code/src/test/java/io/confluent/developer/FlinkSqlFilteringTest.java (hunk 2 above)
Comment (bbejeck) [reply _includes/tutorials/filtering/flinksql/code/src/test/java/io/confluent/developer/FlinkSqlFilteringTest.java:27]: good catch -fixed

**Comment by bbejeck** [comment]
addressed all comments and [unit test passes ](https://semaphore.ci.confluent.io/jobs/3407d761-5ad4-4cb8-8338-38711260dcb4)

**Pull request merged**